pkg/apis/core/validation/validation.go

```diff
@@ -1321,6 +1321,32 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 	return allErrs
 }
 
+// ValidateMountRecursiveReadOnly validates RecursiveReadOnly mounts.
+func ValidateMountRecursiveReadOnly(mount core.VolumeMount, fldPath *field.Path) field.ErrorList {
+	if mount.RecursiveReadOnly == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
```

Comment:
You don't need to check the gate here - if the dropDisabled logic left the field in place, we validate it.

Reply:
fixed